grading/load_config_json.cpp

```diff
@@ -20,6 +21,52 @@ void json_set_default(nlohmann::json &whole_config, std::string field, const T&
   }
 }
 
+uint64_t parse_file_size(std::string size_str) {
+  // Trim whitespaces, it cannot handle UTF-8 characters
+  size_str.erase(std::remove_if(size_str.begin(), size_str.end(), isspace), size_str.end());
+
+  // Convert string to lowercase, it cannot handle UTF-8 characters
+  for (char& c : size_str)
+    c = std::tolower(c);
+
+  // Accepted units, all Bs could be omitted:
+  //    Base 1000: B, KB,  MB,  GB,  TB   (SI)
+  //    Base 1024: B, KiB, MiB, GiB, TiB  (IEC-i)
+  const std::regex accepted { "\\d+([kmgt]i?)?b?" };
+
+  // Extract numeric file size, will cause exception if the number is larger than 64-bit
+  size_t unit_begin  = 0;
+  uint64_t file_size = std::stoull(size_str, &unit_begin);
+  // check if it is 0
+  if (file_size == 0) {
```

Comment:
Can we use the regex to test whether the file_size is 0 without having a seperate check for that. Is there a reason to have a seperate check for this. May a regex like `[1-9]\d*([kmgt]i?)?b?` this.(Not sure whether this is the correct one.)

Reply:
Thank you, I think this is the correct one

Reply:
Sorry I just noticed this. Something like `0123` is going to fail using this regex. Not sure whether this should be accepted.

Reply:
Yes, when the number starts with `0`, it is considered as base 8 for `stoull`.  As we are only dealing with base 10, I think it should be fine here.

Reply:
I tested this. `stoull` for me seems to be giving the output without `0`( eg. `0123` gives `123`)

Reply:
Maybe try 077?

Reply:
That also gives me 77.

Reply:
Did a google search `stoull` default base is base 10. You can change it using the 3rd parameter. I guess you can go with the previous regex and the check for file size zero. Sorry about that change

Reply:
Oh, my bad, from the documentation, it uses base 10 by default...  I will revert the changes.

Reply:
Thanks, sorry again for the confusion